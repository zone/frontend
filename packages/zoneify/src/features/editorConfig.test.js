const { addFile } = require('../install')
const { name, questions, run } = require('./editorConfig')

jest.mock('../install')

describe('features/editorConfig', () => {
  it('has a name', () => {
    expect(name).toMatchSnapshot()
  })

  it('has no questions', () => {
    expect(questions).toMatchSnapshot()
  })

  describe('run', () => {
    run()

    it('has a run function', () => {
      expect(typeof run).toBe('function')
    })

    it('adds a file', () => {
      expect(addFile).toMatchSnapshot()
    })
  })
})
