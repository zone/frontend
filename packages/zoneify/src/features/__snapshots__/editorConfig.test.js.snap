// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features/editorConfig has a name 1`] = `"EditorConfig"`;

exports[`features/editorConfig has no questions 1`] = `undefined`;

exports[`features/editorConfig run adds a file 1`] = `
[MockFunction addFile] {
  "calls": Array [
    Array [
      Object {
        "content": "# editorconfig.org
root = true

[*]
indent_style = space
indent_size = 2
end_of_line = lf
charset = utf-8
trim_trailing_whitespace = true
insert_final_newline = true

[*.md]
trim_trailing_whitespace = false",
        "fileName": ".editorConfig",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
